### PR TITLE
Detect duplicate measurement keys #1687

### LINKED ISSUE #2152: Duplicate measurement keys don't raise exception
Found by @Strilanc in review of #2114 .

To reproduce:

```
In [1]: import cirq

In [2]: q = cirq.NamedQubit('q')

In [3]: c = cirq.Circuit.from_ops(cirq.measure(q, key='a'), cirq.measure(q, key='a'))

In [4]: c
Out[4]: q: ───M('a')───M('a')───

In [5]: sim = cirq.Simulator()

In [6]: sim.run(c, repetitions=5)
Out[6]: a=0000000000
```

### PR DESCRIPTION
**Pull request by iamvamsikrishnad**
Fixes #2152.

### CODE REVIEW

**Comment by vtomole** [comment]
@iamvamsikrishnad Please fix the merge conflicts.

File: cirq/google/xmon_device.py [hunk 1]
```diff
@@ -215,17 +212,5 @@ def __str__(self):
...
-def _verify_unique_measurement_keys(operations: Iterable[ops.Operation]):
-    seen = set()  # type: Set[str]
-    for op in operations:
-        if protocols.is_measurement(op):
-            key = protocols.measurement_key(op)
-            if key in seen:
-                raise ValueError('Measurement key {} repeated'.format(key))
-            seen.add(key)
+        return (self._measurement_duration, self._exp_w_duration,
+                self._exp_z_duration, self.qubits)
```
Comment (Strilanc) [inline cirq/google/xmon_device.py]: I'm surprised the formatter lets lack of trailing newlines through

File: cirq/google/xmon_device_test.py [hunk 2]
```diff
@@ -227,31 +227,6 @@ def test_validate_scheduled_operation_not_adjacent_exp_11_exp_w():
     d.validate_schedule(s)
 
 
-def test_validate_circuit_repeat_measurement_keys():
```
Comment (Strilanc) [inline cirq/google/xmon_device_test.py]: Why was this test removed with no test being added to e.g. circuit_test.py to compensate? What is checking that we detect dupes?

**Comment by vtomole** [comment]
Hey @iamvamsikrishnad. I want to get this in. Please address the requested changes.

**Comment by iamvamsikrishnad** [comment]
@vtomole : I will take a look at this after 2 weeks, as I busy with something else now. Please let me know if you need it in a shorter time.

**Comment by vtomole** [comment]
@iamvamsikrishnad No problem. Take your time.

**Comment by iamvamsikrishnad** [comment]
I am closing this one and submitting a new PR, as this is quite old and some of the changes I feel are irrelevant.

**Pull request closed without merging**